code/game/mecha/equipment/tools/tools.dm

```diff
@@ -28,7 +28,18 @@
 		if(istype(target,/obj))
 			var/obj/O = target
 			if(!O.anchored)
-				if(cargo_holder.cargo.len < cargo_holder.cargo_capacity)
+				var/obj/structure/ore_box/ore_box = locate(/obj/structure/ore_box) in cargo_holder.cargo
+				if(ore_box && istype(O, /obj/item/weapon/ore))
+					var/count = 0
+					for(var/obj/item/weapon/ore/I in get_turf(target))
+						if(I.material)
+							ore_box.materials.addAmount(I.material, 1)
+							qdel(I)

```

Comment:
I'll ask you about this in the IRC later since I'm not sure about this pooling business and don't want to fuck anything up